google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -94,11 +119,78 @@ def log_metrics(self, metrics: Dict[str, Union[str, float, int]]):
         )
         artifact.update(metadata=metrics)
 
-    def get_experiment(self, experiment: str):
-        raise NotImplementedError("get_experiment not implemented")
+    def get_experiment(self, experiment: str) -> pd.DataFrame:
+        """Returns a Pandas DataFrame of the parameters and metrics associated with one experiment.
+
+            Example:
+
+            aiplatform.init(experiment='exp-1')
+            aiplatform.set_run(run='run-1')
+            aiplatform.log_params({'learning_rate': 0.1})
+            aiplatform.log_metrics({'accuracy': 0.9})
+
+            aiplatform.set_run(run='run-2')
+            aiplatform.log_params({'learning_rate': 0.2})
+            aiplatform.log_metrics({'accuracy': 0.95})
+
+            Will result in the following DataFrame
+            ___________________________________________________________________________
+            | experiment_name | run_name      | param.learning_rate | metric.accuracy |
+            ---------------------------------------------------------------------------
+            | exp-1           | run-1         | 0.1                 | 0.9             |
+            | exp-1           | run-2         | 0.2                 | 0.95            |
+            ---------------------------------------------------------------------------
+
+            Args:
+              experiment: Name of the Experiment to filter results.
+
+            Returns:
+              Pandas Dataframe of Experiment with metrics and parameters.
+            """
+
+        # TODO remove this extra API call once our BE can support project ID in filters.
+        try:
+            experiment_context = _Context(resource_name=experiment)
```

Comment:
IIRC I was doing the get to avoid catching exceptions in the caller, and just doing something like:

```
experiment_context  = _Context.get(resource_name=experiment)
if not experiment_context:
   ....
```

No strong opinion, this also works

Reply:
Good point, I think get can help remove some of the duplicated codes. But I believe _get was designed to be a private method right now, so probably should not directly be called here? Do you think we should make the get method public?

Reply:
I made it private because I was just using get_or_create, but I think we can make it public, yes, in case you prefer this other style instead of catching exceptions in metadata.py...

Reply:
Updated this part of code based on Sasha's feedback and we decided to raise the not found error instead of catch it. So no need to use the get method here.